### PR TITLE
Refine description of :not() pseudo-class (#18730)

### PR DESCRIPTION
**Pull request by sideshowbarker**
Replaces commit 6dbaf68b4f0cd6b743d9e039a6ce707caf0d7191. See https://github.com/mdn/content/issues/16975#issuecomment-1193350060

### CODE REVIEW

File: files/en-us/web/css/_colon_not/index.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ There are several unusual effects and outcomes when using `:not()` that you shou
...
 - The specificity of the `:not()` pseudo-class is replaced by the specificity of the most specific selector in its comma-separated argument of selectors; providing the same specificity as if it had been written [`:not(:is(argument))`](/en-US/docs/Web/CSS/:is).
 - `:not(.foo)` will match anything that isn't `.foo`, _including {{HTMLElement("html")}} and {{HTMLElement("body")}}._
-- The `:not()` pseudo-class may not work as expected when used with [descendant combinators](/en-US/docs/Web/CSS/Descendant_combinator). For instance, if you do `body :not(table) a`, instead of excluding all the links that are the direct or indirect children of {{HTMLElement("table")}} like you would expect, it excludes only the links that are direct children of {{HTMLElement("table")}}. That means any links you may have inside a {{HTMLElement("tr")}} or {{HTMLElement("td")}} will still be selected.
+- This selector only applies to one element; you cannot use it to exclude all ancestors. For instance, `body :not(table) a` will still apply to links inside a {{HTMLElement("table")}}, since {{HTMLElement("tr")}} will match with the `:not()` part of the selector.
```
Comment (Josh-Cena) [inline files/en-us/web/css/_colon_not/index.md:45]: I wonder if we can still rephrase it somehow to fix https://github.com/mdn/content/issues/16975 properly. The idea of "ancestors" and "children" is certainly ambiguous in this context. What about:

```suggestion
- This selector will match anything that's literally "not an X". This may be surprising when used with [descendant combinators](/en-US/docs/Web/CSS/Descendant_combinator), since there are multiple paths to select a target element. For instance, `body :not(table) a` will still apply to links inside a {{HTMLElement("table")}}, since {{HTMLElement("tr")}} will match with the `:not()` part of the selector.
```

It kind of coincides with the previous point—I think we can merge them.

File: files/en-us/web/css/_colon_not/index.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ There are several unusual effects and outcomes when using `:not()` that you shou
...
 - `:not(.foo)` will match anything that isn't `.foo`, _including {{HTMLElement("html")}} and {{HTMLElement("body")}}._
-- The `:not()` pseudo-class may not work as expected when used with [descendant combinators](/en-US/docs/Web/CSS/Descendant_combinator). For instance, if you do `body :not(table) a`, instead of excluding all the links that are the direct or indirect children of {{HTMLElement("table")}} like you would expect, it excludes only the links that are direct children of {{HTMLElement("table")}}. That means any links you may have inside a {{HTMLElement("tr")}} or {{HTMLElement("td")}} will still be selected.
+- This selector will match anything that's literally "not an X". This may be surprising when used with [descendant combinators](/en-US/docs/Web/CSS/Descendant_combinator), since there are multiple paths to select a target element. For instance, `body :not(table) a` will still apply to links inside a {{HTMLElement("table")}}, since {{HTMLElement("tr")}} will match with the `:not()` part of the selector.
```
[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/web/css/_colon_not/index.md:45]: adding a few more elements to encourage understanding how complex a selector would need to be to target everything but table links.

[the remark this answers is not in the document]
Comment (edbaafi) [reply files/en-us/web/css/_colon_not/index.md:45]: @Josh-Cena @estelle Thanks this looks much better!  

Nitpicking here but I wonder if we can avoid the word `is` as in `This selector will match everything that is "not an X"` as `is` is not the most general way to describe the `complex-selector-list` as it only really applies to simple element selectors (e.g. we can say an element `is not` a table but not that it `is not` a class of ".foo" because elements `have` classes/attributes/ids but `are` not those classes/attributes/ids. Perhaps we can reuse the language from the first line of the docs:  `The selector represents elements that do not match a list of selectors.  This may be surprising when...`  

I know that this particular example does use a simple element selector (table) but the point of the example is to illustrate the broader issue which is not specific to simple element selectors.
 
TL;DR
  
When considering `is` vs `matches`, I am aware there is an `:is()` psuedo-class [that replaced](https://github.com/w3c/csswg-drafts/issues/3258) `:matches()` with the primary argument being that `is` and `not` are semantically opposites.  But this justification was misguided as `is` is a verb and `not` is an adverb and these are not clear semantic pairs as was claimed.  The fact these are not opposites is clear when you consider that we can say `is not` where the adverb `not` modifies the verb `is`.  `:not()` and `:matches()` were in fact a perfectly fine pairing where `not` can be implied to mean `does not match` which is the opposite of `matches`.

[the remark this answers is not in the document]
Comment (sideshowbarker) [reply files/en-us/web/css/_colon_not/index.md]: > Nitpicking here but I wonder if we can avoid the word `is` as in `This selector will match everything that is "not an X"`

I’ll leave it up to @estelle to make that call on whether to make that or other further changes in this PR branch

**Review by estelle — APPROVED** [review]
